pkg/cache/groupcache.go

```diff
@@ -135,6 +132,12 @@ func NewGroupcacheWithConfig(logger log.Logger, reg prometheus.Registerer, conf
 
 			switch parsedData.Verb {
 			case cachekey.AttributesVerb:
+				_, attrCfg := cfg.FindAttributesConfig(parsedData.Name)
+				if attrCfg == nil {
+					// TODO: Debug this. Why? Attributes get called for Chunks.
+					panic("caching bucket layer must not call on unconfigured paths")
+				}
+
 				attrs, err := bucket.Attributes(ctx, parsedData.Name)
```

Comment:
In this branch, we can now also use `Codec.Encode()` here instead of `json.Marshal`

Reply:
Do you mean `attrCfg.Codec.Encode()`? If yes, it won't work for all the config types we have here. `Codec` exists for `iterCfg` but not for `attrCfg`.

Reply:
Let's leave `json.Marshal` for now, this could be changed in the future :smile: